src/mongo/db/commands/index_filter_commands.cpp

```diff
@@ -167,8 +167,8 @@ namespace mongo {
         // This is a read lock. The query settings is owned by the collection.
         Client::ReadContext readCtx(ns);
         Client::Context& ctx = readCtx.ctx();
-        QuerySettings* querySettings;
-        PlanCache* unused;
+        QuerySettings* querySettings = NULL;
```

Comment:
This shouldn't be necessary. What warning or error is GCC emitting? Pointers to these are passed into getQuerySettingsAndPlanCache right below.